fix: Use function parameters instead of gflags for numDrivers and numWorkers in unittests

MBkkt opened the pull request:
Also
* test more cases and have better output in case of failure
* remove unnecessary namespaces

mbasmanova commented on axiom/optimizer/tests/QueryTestBase.cpp:
```diff
@@ -191,15 +191,6 @@ std::shared_ptr<core::QueryCtx> QueryTestBase::getQueryCtx() {
   return queryCtx_;
 }
 
-optimizer::PlanAndStats QueryTestBase::planVelox(
```
I assume some call sites are affected by this change. Should we identify these and make sure they continue to work as before?

MBkkt replied:
fixed, (default value is same as it was before my patch)

mbasmanova commented on axiom/optimizer/tests/SubfieldTest.cpp:
```diff
@@ -28,7 +28,7 @@
 DECLARE_uint32(optimizer_trace);
 
-DECLARE_int32(num_workers);
+DECLARE_uint32(num_workers);
```
Shouldn't we delete this and change the code in toSingleNodePlan to pass 'options'?

```
core::PlanNodePtr toSingleNodePlan(
      const lp::LogicalPlanNodePtr& logicalPlan) {
    gflags::FlagSaver saver;
    FLAGS_num_workers = 1;

    auto plan = planVelox(logicalPlan).plan;
```

mbasmanova commented on axiom/optimizer/tests/TpchPlanTest.cpp:
```diff
@@ -25,7 +25,7 @@
 
 DECLARE_uint32(optimizer_trace);
-DECLARE_int32(num_workers);
+DECLARE_uint32(num_workers);
```
Wondering if we can simply delete this.

MBkkt replied:
Fixed

MBkkt commented:
@mbasmanova I applied your suggestions and rebased

MBkkt commented:
@mbasmanova I removed from this PR anything that relates to this issue https://github.com/facebookexperimental/verax/issues/413 and updated PR description

Is it ok now?

mbasmanova commented on axiom/optimizer/tests/SubfieldTest.cpp:
```diff
@@ -364,11 +362,11 @@ class SubfieldTest : public QueryTestBase,
-
-    auto plan = planVelox(logicalPlan).plan;
+      const lp::LogicalPlanNodePtr& logicalPlan,
+      int32_t numDrivers = 1) {
```
Do we have any callers that specify this argument? If not, perhaps, we can remove it.

mbasmanova commented on axiom/optimizer/tests/PlanTest.h:
```diff
@@ -76,25 +76,43 @@ class PlanTest : public test::QueryTestBase {
-
-    if (options.numWorkers != 1) {
-      auto singleNodePlan = planVelox(
+    auto referenceResult = runVelox(referencePlan);
```
I didn't realize we have PlanTest.h. Would it be possible to delete this file and move the code to .cpp?

MBkkt replied:
Ok.
